.circleci/config.yml

```diff
@@ -112,16 +112,17 @@ references:
                   name: Build Workspace | << parameters.workspace >>
                   working_directory: << parameters.workspace >>
                   command: |
-                    if [ -d install ] && [ ! -f build_failed ]
+                    BUILD_UNFINISHED=$(colcon list --packages-skip-build-finished)
+                    BUILD_FAILED=$(colcon list --packages-select-build-failed)
+                    if [ -n "$BUILD_UNFINISHED" ] || [ -n "$BUILD_FAILED" ]
                     then
-                      echo "Skipping Build"
-                    else
```

Comment:
this was something that was confusing to me with the old way because "install" always exists in the overlay because it is a simlink that is created in before_checkout

Reply:
The symlink install folder is only created for emulating a conventional workspace by using the installed ROS distro. This build command above only runs for the underlay and overlay workspaces, not the emulated distro workspace.